### PR TITLE
feat: add initial Popover component implementation

### PR DESCRIPTION
**Pull request by web-padawan**
## Description

Part of https://github.com/vaadin/platform/issues/5271

The following logic is implemented:

- adding children 
  - the logic and `PopoverChildrenTest` copied from `Dialog`
- implementing `HasStyle`
  - the logic and `PopoverHasStyleTest` copied from `Dialog`
- `setTarget()` and `getTarget()`
  - including a logic that checks for target detach / reattach
- `setFor()` and `getFor()`
- `setPosition()` and `getPosition()`
  - the `PopoverPosition` enum is a copy of `TooltipPosition`

## Type of change

- Feature

### CODE REVIEW

File: vaadin-popover-flow-parent/vaadin-popover-flow/src/main/java/com/vaadin/flow/component/popover/Popover.java [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+        if (target instanceof Text) {
+            throw new IllegalArgumentException(
+                    "Text as a target is not supported. "
+                            + "Consider wrapping the Text inside a Div.");
+        }
+
+        this.target = target;
+
+        if (target == null) {
+            getElement().executeJs("this.target = null");
```
Comment (vursen) [inline vaadin-popover-flow-parent/vaadin-popover-flow/src/main/java/com/vaadin/flow/component/popover/Popover.java:135]: It seems that the attach listener is currently not removed when the target is disconnected from the popover.

File: vaadin-popover-flow-parent/vaadin-popover-flow/src/main/java/com/vaadin/flow/component/popover/Popover.java [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+        // because Flow creates a new client-side element
+        target.getUI().ifPresent(this::onTargetAttach);
+        targetAttachRegistration = target
+                .addAttachListener(e -> onTargetAttach(e.getUI()));
+    }
+
+    private void onTargetAttach(UI ui) {
+        if (target != null) {
+            ui.getPage().executeJs("$0.target=$1", getElement(),
+                    target.getElement());
```
Comment (vursen) [inline vaadin-popover-flow-parent/vaadin-popover-flow/src/main/java/com/vaadin/flow/component/popover/Popover.java:149]:
```suggestion
            getElement().executeJs("this.target = $0", target.getElement());
```

File: vaadin-popover-flow-parent/vaadin-popover-flow/src/main/java/com/vaadin/flow/component/popover/Popover.java [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+    @Override
+    public void addComponentAtIndex(int index, Component component) {
+        HasComponents.super.addComponentAtIndex(index, component);
+
+        updateVirtualChildNodeIds();
+    }
+
+    private void attachComponentRenderer() {
+        this.getElement().executeJs(
+                "Vaadin.FlowComponentHost.patchVirtualContainer(this)");
```
Comment (vursen) [inline vaadin-popover-flow-parent/vaadin-popover-flow/src/main/java/com/vaadin/flow/component/popover/Popover.java:203]:
```suggestion
        getElement().executeJs(
                "Vaadin.FlowComponentHost.patchVirtualContainer(this)");
```

File: vaadin-popover-flow-parent/vaadin-popover-flow/pom.xml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>jakarta.servlet</groupId>
+            <artifactId>jakarta.servlet-api</artifactId>
+        </dependency>
```
[the remark this answers is not in the document]
Comment (web-padawan) [reply vaadin-popover-flow-parent/vaadin-popover-flow/pom.xml:47]: I think it's just copied from somewhere.

File: vaadin-popover-flow-parent/vaadin-popover-flow/src/main/java/com/vaadin/flow/component/popover/Popover.java (hunk 1 above)
Comment (web-padawan) [reply vaadin-popover-flow-parent/vaadin-popover-flow/src/main/java/com/vaadin/flow/component/popover/Popover.java:135]: Thanks, fixed.

File: vaadin-popover-flow-parent/vaadin-popover-flow/src/main/java/com/vaadin/flow/component/popover/Popover.java (hunk 2 above)
Comment (web-padawan) [reply vaadin-popover-flow-parent/vaadin-popover-flow/src/main/java/com/vaadin/flow/component/popover/Popover.java:149]: Done.

File: vaadin-popover-flow-parent/vaadin-popover-flow-integration-tests/src/test/java/com/vaadin/flow/component/popover/tests/PopoverIT.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+@TestPath("vaadin-popover")
+public class PopoverIT extends AbstractComponentIT {
+
+    static final String POPOVER_OVERLAY_TAG = "vaadin-popover-overlay";
+
+    @Before
+    public void init() {
+        open();
+        waitUntil(driver -> findElements(By.tagName("vaadin-popover"))
+                .size() > 0);
```
[the remark this answers is not in the document]
Comment (web-padawan) [reply vaadin-popover-flow-parent/vaadin-popover-flow-integration-tests/src/test/java/com/vaadin/flow/component/popover/tests/PopoverIT.java]: Removed. I added this before to try to stabilize the test that failed in CI but passed locally but it was a wrong fix.

File: vaadin-popover-flow-parent/vaadin-popover-flow-integration-tests/src/test/java/com/vaadin/flow/component/popover/tests/PopoverIT.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        clickTarget();
+        checkPopoverIsOpened();
+
+        waitForElementPresent(By.id("popover-content"));
+    }
+
+    @Test
+    public void detachAndReattachTarget_clickTarget_popoverOpensAndCloses() {
+        $(NativeButtonElement.class).id("detach-target").click();
+        $(NativeButtonElement.class).id("attach-target").click();
```
[the remark this answers is not in the document]
Comment (web-padawan) [reply vaadin-popover-flow-parent/vaadin-popover-flow-integration-tests/src/test/java/com/vaadin/flow/component/popover/tests/PopoverIT.java]: Updated to use `clickElementWithJs` for consistency.

File: vaadin-popover-flow-parent/vaadin-popover-flow/src/main/java/com/vaadin/flow/component/popover/Popover.java [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+        getElement().setProperty("position", position.getPosition());
+    }
+
+    /**
+     * Gets position of the popover with respect to its target.
+     *
+     * @return the position
+     */
+    public PopoverPosition getPosition() {
+        var positionString = getElement().getProperty("position");
```
[the remark this answers is not in the document]
Comment (web-padawan) [reply vaadin-popover-flow-parent/vaadin-popover-flow/src/main/java/com/vaadin/flow/component/popover/Popover.java]: Done.

File: vaadin-popover-flow-parent/vaadin-popover-flow/src/main/java/com/vaadin/flow/component/popover/Popover.java [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+     *            the target component for this popover, can be {@code null} to
+     *            remove the target
+     * @throws IllegalArgumentException
+     *             if the target is a {@link Text} component.
+     */
+    public void setTarget(Component target) {
+        if (target instanceof Text) {
+            throw new IllegalArgumentException(
+                    "Text as a target is not supported. "
+                            + "Consider wrapping the Text inside a Div.");
```
[the remark this answers is not in the document]
Comment (web-padawan) [reply vaadin-popover-flow-parent/vaadin-popover-flow/src/main/java/com/vaadin/flow/component/popover/Popover.java]: Done.

**Review by tomivirkki — APPROVED** [review]

**Pull request merged**